gh-140594: Fix an out of bounds read when feeding NUL byte to `PyOS_StdioReadline` (#140910)
python/cpython: Misc/NEWS.d/next/Core_and_Builtins/2025-11-02-16-23-17.gh-issue-140594.YIWUpl.rst

> Fix a buffer overflow when a single NULL character is read from the standard input.

Reviewer: I'm not sure if this PR should be considered as Security fix. Maybe @picnixz can shed some light on this.

Reviewer 2: Oh yeah it's not the right place. It's a simple bug fix. The attack surface is too small IMO for it to be considered a vuln. The function is not exposed directly to the user and the interpreter doesn't even crash at runtime I think.

Author: fixed

> Fix a buffer overflow when a single NULL character is read from the standard input.

Reviewer 3: It's not a buffer overflow, "just" a read out of bounds.

```suggestion
Fix an out of bounds read when a single NUL character is read from the standard input.
```
